andela/eagle-bn-backend
# #168781808 implement user password reset

#### What does this PR do?
- Implements user password reset incase a user forgot his password, where a notification is sent to user email for reset instructions
#### Description of Task to be completed?
- validate input email address
- install `nodemailer` package
- add new data in `.env sample`
- implement email sender
- validate password on reset
- confirm password
#### How should this be manually tested?
- `npm i `to install relevant package added `nodemail`
- add new environments in `.env file`
```
MAIL_HOST:                 #[like gmail.com]
MAIL_PORT:                 #[any port can be null or excluded]
MAIL_SENDER:               #[any sender barfoot nomad eagels]
MAIL_USER:                 #[any email sender example@gmail.com]
MAIL_PASSWORD:             #[any email sender password]
MAIL_SERVICE:              #[like gmail]

```
- add new packages
- access `hostaddress/api/v1/users/reset-password`
- provide the email to be reset in the body.
- notification and instructions are sent to the email provided.
- a link redirects you requests to add `new password` and confirm it
#### Any background context you want to provide?
**Basic analysis design**
<img width="842" alt="Screen Shot 2019-10-12 at 09 24 05" src="https://user-images.githubusercontent.com/41104288/66701126-fc45fb00-ecf8-11e9-836a-c90d31d5047e.png">

#### What are the relevant pivotal tracker 
[#168781808](https://www.pivotaltracker.com/story/show/168781808)
#### Screenshots (if appropriate)
<img width="1121" alt="Screen Shot 2019-10-10 at 17 04 30" src="https://user-images.githubusercontent.com/41104288/66581239-101a2180-eb80-11e9-9229-9d7d9ac4c9b8.png">

<img width="1440" alt="Screen Shot 2019-10-10 at 16 58 27" src="https://user-images.githubusercontent.com/41104288/66580922-76eb0b00-eb7f-11e9-8d6e-ab038800291e.png">
<img width="1128" alt="Screen Shot 2019-10-11 at 12 10 16" src="https://user-images.githubusercontent.com/41104288/66643982-2d9ec800-ec20-11e9-858e-b9fe5ffae00a.png">


#### Questions:
na
